software/main.py

```diff
@@ -104,198 +136,232 @@ def main_loop():
+                    targeted_ball = processed_Data.balls[-1]
+                    x_cord = targeted_ball.x
+                    y_cord = targeted_ball.y
+                    dist = targeted_ball.distance
+                    delta = ((x_cord-reso_x_mid)/reso_x_mid)
+                    
+                    
+                    #Follows where was the ball last so find_ball function is more efficient
                     if delta<0:
                         ball_right_side = 0
                     else:
                         ball_right_side = 1
                     
-                        #SpeedY based on ball distance
-                    if dist > 250:
-                        speed_Y=0.25
-                    elif dist > 325:
-                        speed_Y=0.15
-                    else:
-                        speed_Y=1.2
-                        #Controlls that balls location is ready for robot's orbit function
-                    if x_cord < (reso_x_mid + 5) and x_cord >(reso_x_mid -5) and dist > 385:
-                        #robot.center_ball(xcord)
+                    #SpeedY based on ball distance
+
+                    speed_Y = calculator.sig_approach(y_cord,max_move_Yspeed, change_move_Y)
+                        
+                    #Controlls that balls location is ready for robot's orbit function
+                    if x_cord < (reso_x_mid + 150) and x_cord >(reso_x_mid -150) and dist > 355:
                         state=State.ORBIT
+                          
                     else:
                         #Moves to ball
-                        speed_R = delta*(-3.5)
-                        speed_X = delta*0.5
+
+                        speed_R = -(calculator.sig_correction_move(x_cord,max_move_Rspeed, change_move_R))
+                        speed_X = calculator.sig_correction_move(x_cord,max_move_Xspeed, change_move_X)
                         robot.move(speed_X, speed_R, speed_Y, speed_T)
+                        print("x speed on - ", speed_X)
+                        print("y speed on - ", speed_Y)
+                        print("r speed on - ", speed_R)
+
                 else:
                     #If there is no ball
-                    state=State.FIND_BALL
+                    state = State.FIND_BALL
                         
             elif state==State.ORBIT:
                 print("--------Centering ball and basket--------")
                 #If there is ball
-                if len(processed_Data.balls)>0:
+                if len(processed_Data.balls) > 0:
                     targeted_ball=processed_Data.balls[-1]
                     x_cord=targeted_ball.x
                     y_cord=targeted_ball.y
                     dist=targeted_ball.distance
-                    delta=((x_cord-reso_x_mid)/reso_x_mid)
 
                     #controlls basket colour
-                    if dist<220:
+                    if dist < 340:

```

Comment:
Maybe use a ration of camera resolution here

Reply:
Added an orbit threshold variable.